AxonDeepSeg/segment.py

```diff
@@ -19,313 +18,197 @@
-    model = json.load(open(path_model / model_json[0]))
+def get_model_input_format(path_model: Path) -> tuple[str, int]:
+    '''
+    Get the input format of the model used for segmentation (e.g. '.png')
 
-    model_resolution = [model['transformation']['Resample']['wspace'], model['transformation']['Resample']['hspace']]
-    model_resolution[0] = model_resolution[0] * resolution_unit_conversion_factor
-    model_resolution[1] = model_resolution[1] * resolution_unit_conversion_factor
-    
-    patch_size = [model['default_model']['length_2D'][0], model['default_model']['length_2D'][1]]
+    Parameters
+    ----------
+    path_model : Path
+        Path to the folder containing the model.
     
-    exception_msg = "This model uses an anisotropic resolution, which AxonDeepSeg does not yet support."
-
-    if model_resolution[0] == model_resolution[1]: #isotropic pixels
-        model_resolution = model_resolution[0]
-    else:
-        raise Exception(exception_msg)
+    Returns
+    -------
+    (format, n_channels): tuple[str, int]
+        model input format (e.g. '.png') and nb. of input channels
+    '''
+    with open(path_model / 'dataset.json') as f:
+        dataset_dict = json.load(f)
+    fmt = dataset_dict['file_ending']
+    channels = list(dataset_dict['channel_names'].keys())
+    return fmt, len(channels)
 
-    if patch_size[0] == patch_size[1]: #isotropic pixels
-        patch_size = patch_size[0]
-    else:
-        raise Exception(exception_msg)
+@logger.catch
+def prepare_inputs(path_imgs: List[Path], file_format: str, n_channels: int) -> List[Path]:
+    '''
+    Verifies if the input images can be sent to axon_segmentation(). Otherwise, 
+    converts and saves in expected format.
+
+    Parameters
+    ----------
+    path_imgs : List(Path)
+        List of paths to the images to prepare.
+    file_format : str
+        Expected file format for the images, e.g. '.png' or '.tif'.
+    n_channels : int
+        Number of channels expected by the model, e.g. 1 for grayscale.
+
+    Error
+    -----
+    If n_channels > 1 and the image doesn't have enough channels, an error is raised.
+    Note that if n_channels == 1, the image will be converted to grayscale if needed.
+
+    Returns
+    -------
+    filelist : List(Path)
+        List of paths to sanitized images. If an image is already in the 
+        expected format, the path is the same as the one in path_imgs. Otherwise, 
+        will be the path to the converted image.
+    '''
+    filelist = []
+    for im_path in path_imgs:
+        target = im_path
+
+        imshape = get_imshape(str(target)) # HWC format
+        is_correct_shape = imshape[-1] == n_channels
+        is_correct_format = target.suffix == file_format
+        needs_conversion = not is_correct_shape or not is_correct_format
+        
+        if not is_correct_shape and n_channels != 1:
+            logger.error(f'{str(target)} has {imshape[-1]} channels, expected {n_channels}.')
```

Comment:
self: 2 lines not covered by tests

Reply:
This `if` condition might (?) miss what I think is an edge case, but overall could be improved the by simplifying the "readability" of the condition.

If `imshape[-1] == 3` (or 4), but `n_channels == 1), then `is_correct_shape == False`, but `n_channels != 1` is True. Thus `(not is_correct_shape) and (n_channels != 1)` (parentheses mine) will be False, @hermancollin is this expected behaviour? I read further on but am not sure if this case should "pass"

Reply:
Lines covered by tests introduced in 1760ad7

Reply:
great change, this part is more clear now